exercises/simple-linked-list/test/simple_linked_list_tests.erl

```diff
@@ -0,0 +1,91 @@
+-module(simple_linked_list_tests).
+
+-include_lib("erl_exercism/include/exercism.hrl").
+
+-include_lib("eunit/include/eunit.hrl").
+
+'1_value_reflects_constructor_arg_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(2, L),
+    ?assertMatch(2, (simple_linked_list:head(L2))).
+
+'2_new_lists_should_have_length_0_test'() ->
+    L = simple_linked_list:empty(),
+    ?assertMatch(0, (simple_linked_list:count(L))).
+
+'3_adding_a_element_increments_length_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(1, L),
+    ?assertMatch(1, (simple_linked_list:count(L2))).
+
+'4_adding_two_elements_increments_twice_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(1, L),
+    L3 = simple_linked_list:cons(2, L2),
+    ?assertMatch(2, (simple_linked_list:count(L3))).
+
+'5_new_lists_have_a_null_head_element_test'() ->
+    L = simple_linked_list:empty(),
+    ?assertError(_, (simple_linked_list:head(L))).
+
+'6_adding_an_element_to_an_empty_list_sets_the_head_element_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(1, L),
+    ?assertMatch(1, (simple_linked_list:head(L2))).
+
+'7_adding_a_second_element_updates_the_head_element_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(1, L),
+    L3 = simple_linked_list:cons(2, L2),
+    ?assertMatch(2, (simple_linked_list:head(L3))).
+
+'8_can_get_the_next_element_from_the_head_test'() ->
+    L = simple_linked_list:empty(),
+    L2 = simple_linked_list:cons(1, L),
+    L3 = simple_linked_list:cons(2, L2),
+    L4 = simple_linked_list:tail(L3),
+    ?assertMatch(1, (simple_linked_list:head(L4))).
+
+'9_with_correct_length_test'() ->
+    A = array:from_list(lists:seq(1, 10)),
+    L = simple_linked_list:from_array(A),
+    ?assertMatch(10, (simple_linked_list:count(L))).
```

Comment:
I'd prefer to start with converting an empty list rather than a filled one, just to guide the student through the implementation of `from_native_list/1` (see my comment above about the name!)